compiler/src/dotty/tools/dotc/core/Symbols.scala

```diff
@@ -516,6 +516,18 @@ object Symbols {
     /** The current name of this symbol */
     final def name(implicit ctx: Context): ThisName = denot.name.asInstanceOf[ThisName]
 
+    def isHigherOrderTypeParameter(implicit ctx: Context): Boolean = this.maybeOwner.isTypeParameterOrSkolem
+    def isTypeParameterOrSkolem(implicit ctx: Context): Boolean = this.isTypeParam
```

Comment:
Just use `isTypeParam`. `isTypeParameterOrSkolem` would be confusing in a dotty context.

Reply:
I kept the original name to ease review. I don't really know what `Skolem` refers to, and was afraid to lose the change in the port from scalac.